### ISSUE TITLE
How many epochs do we need to train in order to get the accuracy of the given pretrained model?

### ISSUE THREAD

**Issue by kaiser-hamid-rabbi** [kenziyuliu/MS-G3D#17]
I am training the model from scratch with a new custom action dataset. The accuracy was increasing (highest ~ 33.08) and loss was decreasing till 70 epochs. After that, the loss is decreasing but accuracy is decreasing as well and currently, after 100 epochs it is 31.29. Would you please tell me that, in order to get satisfactory accuracy and loss, how many epochs should we train? Thanks in advance!
![Screenshot from 2020-08-05 17-45-54](https://user-images.githubusercontent.com/24420353/89409286-a76de800-d743-11ea-910f-5a89e653b24c.png)

**Comment by kenziyuliu** [comment]
Hi @ZerOne288, thanks for your interest! 

If you are interested in training the model on a new custom dataset, you should tune the learning rate / batch size / epochs / LR scheduler as needed. I would also suggest using tensorboard for visualization. As for the pretrained models on NTU RGB+D 60/120 and Kinetics Skeleton, the training configs can be found in Section 4.2 of the paper. Consider also fine-tuning the pretrained models on your custom dataset instead of training from scratch. 

Hope this helps!

**Comment by kaiser-hamid-rabbi** [comment]
Thank you so much for your kind response. I have already tuned batch size (like 16 samples per GPU) and number of  epochs, but still could not experiment with learning rate. Hopefully, according to section 4.2, I can do that. Also, I am using tensorboard for visualization to follow up the training process.

Could you please tell me, What was the `training loss` of the pre-trained model?
![Screenshot from 2020-08-06 12-04-47](https://user-images.githubusercontent.com/24420353/89497096-75a95f80-d7dd-11ea-93fc-33004b54b8d8.png)
![Screenshot from 2020-08-06 12-04-52](https://user-images.githubusercontent.com/24420353/89497102-793ce680-d7dd-11ea-8883-1089bdb0f19e.png)

**Comment by kenziyuliu** [comment]
Hi @ZerOne288,

The loss at the last epoch for 
- NTU RGB+D 60 Cross Subject Joint is ~0.03 for training and ~0.4 for testing
- NTU RGB+D 60 Cross View Joint is ~0.03 for training and ~0.2 for testing
- NTU RGB+D 120 Cross Subject Joint is ~0.03 for training and ~0.7 for testing
- NTU RGB+D 120 Cross Setup Joint is ~0.03 for training and ~0.6 for testing
- Kinetics Skeleton Joint is ~0.83 for training and ~3.1 for testing

I would kindly add that the training/testing losses for the NTU 60/120 & Kinetics datasets are not correlated with the losses on your custom datasets. 

Hope this helps!
